### PR TITLE
REF: rename PandasArray->NumpyExtensionArray

### LINKED ISSUE #53694: API: rename PandasArray -> NumpyExtensionArray
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
NA
```

### Issue Description

More neutral naming scheme analogous to ArrowExtensionArray.

### Expected Behavior

NA

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #53694 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

I'm open to suggestions on names.

### CODE REVIEW

File: pandas/core/arrays/numpy_.py [hunk 1]
```diff
@@ -131,14 +133,14 @@ def _from_sequence(
...
-    def _from_backing_data(self, arr: np.ndarray) -> PandasArray:
+    def _from_backing_data(self, arr: np.ndarray) -> NumpyExtensionArray:
         return type(self)(arr)
 
     # ------------------------------------------------------------------------
     # Data
 
     @property
-    def dtype(self) -> PandasDtype:
+    def dtype(self) -> NumpyEADtype:
```
Comment (mroeschke) [inline pandas/core/arrays/numpy_.py:143]: Would be good to note that the returned `dtype` has changed too

Comment (jbrockmendel) [reply pandas/core/arrays/numpy_.py:143]: sure.  thoughts on the name NumpyEADtype?  i was torn between wanting parity with ArrowDtype vs confusion with np.dtype

Comment (mroeschke) [reply pandas/core/arrays/numpy_.py:143]: IMO NumpyEADtype sounds good

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**